File: math/3783_mirror_distance_of_an_integer-Sipe/Python/mirror_distance.py
class Solution:
    def mirrorDistance(self, n: int) -> int:
        reverse = 0
        number = n
        while n > 0:
            reverse = (reverse * 10) + (n % 10)
            n = n // 10
        return abs(number - reverse)
    
if __name__ == "__main__":
    test_cases = [25, 10, 7, 121, 100, 1]

    for n in test_cases:
        print(f"Input:       {n}")
        print(f"Output:      {Solution().mirrorDistance(n)}")
        print("-" * 35)

File: math/3783_mirror_distance_of_an_integer-Sipe/TS/mirrorDistance.ts
export {};

function mirrorDistance(n: number): number {
    let number = n;
    let reverseNumber = 0;
    while(n > 0){
        reverseNumber = (reverseNumber * 10) + (n % 10)
        n = Math.floor(n / 10)
    }
    return Math.abs(number - reverseNumber)
};

const testCases: number[] = [25, 10, 7, 121, 100, 1];

for (const n of testCases) {
    console.log(`Input:       ${n}`);
    console.log(`Output:      ${mirrorDistance(n)}`);
    console.log("-".repeat(35));
}